### PR TITLE
React Group Project

### PR DESCRIPTION
**Pull request by jecode93**
In this milestone we:

- Create React App.
- Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/react_router.md).
- Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- Create **routes** and **view** components: `rockets`, `missions`, `my profile`. Add `dragons` _[only if your team has 3 members]_).
  - Use `<NavLink />` for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).
- Create directories for all Redux state slice files:`rockets`, `missions`, `dragons` _[only if your team has 3 members]_.

#### Redux: Fetch data and update Redux store

- Upon first render fetch data from the SpaceX API endpoints:
  - Rockets: `https://api.spacexdata.com/v4/rockets`
  - Missions: `https://api.spacexdata.com/v3/missions`
  - Dragons: `https://api.spacexdata.com/v4/dragons` _[only if your team has 3 members]_
- Once the data are fetched, we dispatch an action to store the selected data in Redux store:
  - Rockets:
    - id
    - rocket_name
    - description
    - images
  - Missions:
    - mission_id
    - mission_name
    - description
  
#### Render UI:lists

- Use `useSelector()` Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes. i.e.:

```javascript
// get rockets data from the store
const rockets = useSelector((state) => state.rockets)
```
- Render a table with the missions' data.

#### Redux: Write actions and reducers for booking rockets and joining missions

- When a user clicks the "Reserve rocket" button, action dispatch to update the store. 

#### Render UI: conditional components rendering

- Rockets that have already been reserved show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket".
- Missions that the user has joined already show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button

```javascript
{rocket.reserved && (
    // render Cancel Rocket button
)}
```

#### Render UI: My Profile section

- Compose two column layout and list ONLY the rockets reserved and missions joined by the user:
  - Render a list of all joined missions (use `filter()`).
  - Render a list of all reserved rockets (use `filter()`).

#### Technical set up\*\*

- Set up the repository on GitHub and use Gitflow.
- Set up React and Redux.
- Set up testing libraries(React Testing Library and Jest).

### CODE REVIEW

File: src/components/ReservedRockets.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      <ul className="mission-list">
+        {reservedRockets.map((list) => {
+          const { id, name } = list;
+          return <li key={id} className="mission-name">{name}</li>;
+        })}
+      </ul>
+    </sections>
+  );
+};
+
```
Comment (rloterh) [inline src/components/ReservedRockets.jsx:21]: - [ ] Rocket reservation is not working as intended. When rockets are reserved and then a user visits the profile page and back, the reservation check is cleared from the rocket page as well as the profile page.

[rocket.webm](https://github.com/jecode93/react-group-project/assets/12745474/bf0dd3e2-114e-4d67-8f88-d7ae8628342b)

File: src/components/ReservedMissions.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        {reservedMissions.map((mission) => {
+          const { id, name } = mission;
+          return <li key={id} className="mission-name">{name}</li>;
+        })}
+      </ul>
+    </sections>
+  );
+};
+
+export default ReservedMissions;
```
Comment (rloterh) [inline src/components/ReservedMissions.jsx:22]: - [ ] Mission reservation is similarly not working as intended. When a mission are joined and then a user visits the profile or other page and back, the mission status is cleared from the mission page as well as on the profile.

[mission.webm](https://github.com/jecode93/react-group-project/assets/12745474/f470d710-b4fe-45a5-9bd5-e834c799c5e6)

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Dennis & Jean, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🚦 __🏃 🏡 🏃‍♂️ 🏚️ 🌴 🗼

## Highlights 👍  
- [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start) used effectively!
- API fetch is properly implemented!
- Professional ReadMe!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @jecode93, @boniface193, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👏 You did a great job fixing the reservations for the rockets and the missions, the app now works as expected.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jecode93** [comment]
@elisha2kyakpo1 thank you very much for your approval...

**Pull request merged**